src/binding/util.h

```diff
@@ -373,6 +488,82 @@ struct AsyncWorkHandle {
 	}
 };
 
+// Big-endian encoding/decoding helpers for transaction log format
+inline void writeUint64BE(char* buffer, uint64_t value) {
+	buffer[0] = static_cast<char>((value >> 56) & 0xFF);
+	buffer[1] = static_cast<char>((value >> 48) & 0xFF);
+	buffer[2] = static_cast<char>((value >> 40) & 0xFF);
+	buffer[3] = static_cast<char>((value >> 32) & 0xFF);
+	buffer[4] = static_cast<char>((value >> 24) & 0xFF);
+	buffer[5] = static_cast<char>((value >> 16) & 0xFF);
+	buffer[6] = static_cast<char>((value >> 8) & 0xFF);
+	buffer[7] = static_cast<char>(value & 0xFF);
+}
+
+inline void writeUint32BE(char* buffer, uint32_t value) {
+	buffer[0] = static_cast<char>((value >> 24) & 0xFF);
+	buffer[1] = static_cast<char>((value >> 16) & 0xFF);
+	buffer[2] = static_cast<char>((value >> 8) & 0xFF);
+	buffer[3] = static_cast<char>(value & 0xFF);
+}
+
+inline void writeUint16BE(char* buffer, uint16_t value) {
+	buffer[0] = static_cast<char>((value >> 8) & 0xFF);
+	buffer[1] = static_cast<char>(value & 0xFF);
+}
+
+inline void writeUint8(char* buffer, uint8_t value) {
+	buffer[0] = static_cast<char>(value);
+}
+
+inline uint64_t readUint64BE(const char* buffer) {
+	return (static_cast<uint64_t>(static_cast<uint8_t>(buffer[0])) << 56) |
```

Comment:
At some point, maybe we could use `htonl` or `bswap_64` for a single machine operation conversion.